components/Input/InputGroup.tsx

```diff
@@ -0,0 +1,67 @@
+import { FC, useRef } from 'react';
+import styled, { css } from 'styled-components';
+import Input from './Input';
+
+interface IInputGroup extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'width'> {
+	icon?: JSX.Element;
+	width?: string;
+	iconPosition?: 'right' | 'left';
+}
+
+interface IInputWrapper {
+	iconSize?: number;
+	iconPosition: 'right' | 'left';
+}
+
+export const InputGroup: FC<IInputGroup> = ({ icon, iconPosition = 'right', ...rest }) => {
+	const iconRef = useRef<HTMLInputElement>();
+
+	return (
+		<InputWrapper
+			className="inputGroup"
+			iconPosition={iconPosition}
+			iconSize={iconRef?.current?.getBoundingClientRect()?.width}
+		>
+			<StyledInput {...rest} />
+			<StyledIcon ref={iconRef as any}>{icon}</StyledIcon>
+		</InputWrapper>
+	);
+};
+
+const StyledInput = styled(Input)``;
+
+const StyledIcon = styled.div``;

```

Comment:
Necessary? You can use `div` and `Input` directly if they don't use any styling.

Reply:
I am not sure you need it to be wrapped. Does it not work if you just use the Input or a div tag in the bottom section? I dont think this is a huge issue though imo. It is fairly minor to have the extra lines if it is a bit easier to make it work

Reply:
just using a div might be too ambiguous if the icon is a div as I can see below. I think its prob ok to leave it wrapped

Reply:
Just tried out and worked without being wrapped... :)